src/Stratis.Bitcoin/Consensus/ScriptAddressResult.cs

```diff
@@ -0,0 +1,23 @@
+﻿namespace Stratis.Bitcoin.Consensus
+{
+    public class ScriptAddressResult
```

Comment:
How about `ScriptAddressResult` hold only an address, and we make an override of that `ScriptColdStakeAddressResult :  ScriptAddressResult` with the additional addresses, then when calling just check the type.

Reply:
Are you thinking to do this so we don't need to ToString() when calling GetDestinationAddress to when retrieving the address?